SpicesUpdate@claudiux/files/SpicesUpdate@claudiux/applet.js

```diff
@@ -0,0 +1,1096 @@
+
+    this.settings.bindProperty(Settings.BindingDirection.BIDIRECTIONAL,
+      "unprotected_applets",
+      "unprotected_applets",
+      this.populateSettingsUnprotectedApplets,
+      null);
+
+    // Desklets
+    this.settings.bindProperty(Settings.BindingDirection.BIDIRECTIONAL,
+      "check_desklets",
+      "check_desklets",
+      this.on_settings_changed,
+      null);
+
+    this.settings.bindProperty(Settings.BindingDirection.BIDIRECTIONAL,
+      "unprotected_desklets",
+      "unprotected_desklets",
+      this.populateSettingsUnprotectedDesklets,
+      null);
+
+    // Extensions
+    this.settings.bindProperty(Settings.BindingDirection.BIDIRECTIONAL,
+      "check_extensions",
+      "check_extensions",
+      this.on_settings_changed,
+      null);
+
+    this.settings.bindProperty(Settings.BindingDirection.BIDIRECTIONAL,
+      "unprotected_extensions",
+      "unprotected_extensions",
+      this.populateSettingsUnprotectedExtensions,
+      null);
+
+    // Themes
+    this.settings.bindProperty(Settings.BindingDirection.BIDIRECTIONAL,
+      "check_themes",
+      "check_themes",
+      this.on_settings_changed,
+      null);
+
+    this.settings.bindProperty(Settings.BindingDirection.BIDIRECTIONAL,
+      "unprotected_themes",
+      "unprotected_themes",
+      this.populateSettingsUnprotectedThemes,
+      null);
+
+    // General settings
+    this.settings.bindProperty(Settings.BindingDirection.IN,
+      "general_frequency",
+      "general_frequency",
+      this.on_settings_changed,
+      null);
+      this.refreshInterval = 3600*this.general_frequency;
+
+    this.settings.bindProperty(Settings.BindingDirection.IN,
+      "general_warning",
+      "general_warning",
+      this.on_settings_changed,
+      null);
+
+    this.settings.bindProperty(Settings.BindingDirection.IN,
+      "events_color",
+      "events_color",
+      this.on_settings_changed,
+      null);
+
+    this.settings.bindProperty(Settings.BindingDirection.IN,
+      "general_notifications",
+      "general_notifications",
+      this.on_settings_changed,
+      null);
+
+    this.settings.bindProperty(Settings.BindingDirection.IN,
+      "displayType",
+      "displayType",
+      this.on_settings_changed,
+      null);
+
+    this.on_orientation_changed(orientation);
+
+    // Init lists of Spices:
+    this.populateSettingsUnprotectedApplets();
+    this.populateSettingsUnprotectedDesklets();
+    this.populateSettingsUnprotectedExtensions();
+    this.populateSettingsUnprotectedThemes();
+
+    // Dependencies:
+    this.dependenciesMet = this.are_dependencies_installed();
+    if (!this.dependenciesMet) this.refreshInterval = 5;
+
+    // ++ Set up Left Click Menu
+    this.menuManager = new PopupMenu.PopupMenuManager(this);
+    this.menu = new Applet.AppletPopupMenu(this, orientation);
+    this.menuManager.addMenu(this.menu);
+
+    // Default icon color
+    let themeNode = this.actor.get_theme_node();
```

Comment:
This is causing this "critical" error:

```
(cinnamon:8955): St-CRITICAL **: 14:45:08.301: st_widget_get_theme_node called on the widget [0x5595c20c1360 StBoxLayout.applet-box ("Spices Update")] which is not in the stage.
```

because the actor isn't realized in _init. I'm not sure if this actually matters because it still can grab the color, but something worth noting. Maybe St needs to be less noisy.

Reply:
I just removed the get_theme_node call.

Reply:
No more critical error. This is working fine.